lib/providers/messages.dart

```diff
@@ -105,17 +105,16 @@ class Messages with ChangeNotifier {
   }
 
   void recieveMessage(List<int> listMessage) async {
-    List<int> recievedMessage = listMessage.sublist(0, listMessage.length - 1);
-    // int checksum = listMessage.last;
-    // if (!validateChecksum(checksum, recievedMessage)) {
-    //   debugPrint('Checksum incorrect');
-    // }
-    // Here we should check the nodeId in the packet and set accordingly
-    User user = await users.getOrCreate(name: 'Becky', nodeId: 1);
+    if (listMessage.length < 4) {
+      return;
+    }
+    var nodeId = listMessage[2].toString() + listMessage[3].toString();
+    int conversationId = listMessage[1];
+    List<int> recievedMessage = listMessage.sublist(4, listMessage.length);
     String parsedMessage = parseMessage(recievedMessage);
-    print(parsedMessage);
-    if (parsedMessage != '') {
-      addMessage(user.id, 1, parsedMessage, DateTime.now());
+    User user = await users.getOrCreate(name: 'Becky', nodeId: int.parse(nodeId));

```

Comment:
Why is becky hard coded?

Reply:
lmao

Reply:
@DelightedWaif

Reply:
tim is actually taking care of the name/nodeId stuff rn haha